File: core/src/main/java/core/qna/CacheSystem.java
package core.qna;

import core.rag.QuestionDTO;
import core.rag.QuestionTitleDTO;

public interface CacheSystem {
    QuestionTitleDTO[] getRecentQuestionTitles(String category, int startNum, int endNum);
    void putRecentQuestionTitle(String category, QuestionTitleDTO questionTitle);
    QuestionDTO getRecentQuestion(String questionId);
    void putRecentQuestion(String questionId, QuestionDTO question);
}

File: core/src/main/kotlin/core/qna/CacheSystem.kt
package core.qna

import core.rag.QuestionDTO
import core.rag.QuestionTitleDTO

interface CacheSystem {
    fun getRecentQuestionTitles(category: String, offset: Int, limit: Int): Array<QuestionTitleDTO>
    fun putRecentQuestionTitle(category: String, questionTitle: QuestionTitleDTO)
    fun getRecentQuestion(questionId: String): QuestionDTO
    fun putRecentQuestion(questionId: String, question: QuestionDTO)
}